compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -587,13 +604,74 @@ impl<'tcx> TyCtxt<'tcx> {
                 is_soft,
                 span,
                 soft_handler,
+                UnstableKind::Regular,
             ),
             EvalResult::Unmarked => unmarked(span, def_id),
         }
 
         is_allowed
     }
 
+    pub fn check_const_stability(self, def_id: DefId, span: Span, const_kw_span: Span) {
```

Comment:
This function is analogous to `check_optional_stability` but with the logic in `eval_stability_allow_unstable` inlined, and which operating on const stability instead of regular stability.

Reply:
That seems like a comment worth putting into the code.